### DISCUSSION TITLE
Page Dynamic Edit - Comportamento inesperado (semelhante ao reportado no #1126)

### DISCUSSION THREAD

**Discussion by wsteixeira** [po-ui/po-angular#1139] [Tire suas dúvidas] [answered]
Olá,

Estou usando o componente "PoPageDynamicEditComponent" está acontecendo um comportamento semelhante ao reportado no #1126. 

Instruções para simulação:

- Altere o valor do campo "Data" manualmente, ou seja, não clique no ícone de calendário e tecle "TAB", neste contexto o evento de mudança será disparado normalmente; 
- **Atualiza a interface antes do próximo teste;**
- Altere o valor do campo "Data" através do ícone de calendário e tecle "TAB", na primeira vez o evento de mudança **não é disparado como esperado**;
- Altere novamente o valor do campo "Data" através do ícone de calendário e tecle "TAB", a partir da segunda vez o evento de mudança é disparado como esperado;

```
  public readonly fields: Array<PoDynamicFormField> = [
    { property: 'id', label: 'Id', key: true, visible: false },
    { property: 'date', label: 'Data', type: 'date', gridColumns: 2, validate: this.onChangeField.bind(this), required: true },
    { property: 'time', label: 'Hora', type: 'time', gridColumns: 2, validate: this.onChangeField.bind(this), required: true },
  ]

  private onChangeField(changeValue): PoDynamicFormFieldValidation {
    console.log(changeValue);
    if (changeValue.property === 'date') {
      console.log('date');
    } else if (changeValue.property === 'time') {
      console.log('time');
    }
    return {};
  }

```

**Comment by alinelariguet** [comment]
Boa tarde @wsteixeira ,
Consegue simular com o StackBlitz (https://stackblitz.com/edit/po-ui)?

**Comment by wsteixeira** [comment, marked as answer]
Olá,

Aline, segue a simulação.

1. Clique ícone "Calendário" do campo "Data" e insira uma data qualquer;
2. Clique no campo "Hora" e **observe que não foi apresentada nenhuma informação no "log" do console**;
3. Insira uma hora qualquer no campo "Hora";
4. Clique no campo "Local" e observe que foram apresentadas informações referentes ao campo "Hora" no log do console;
5. Clique ícone "Calendário" do campo "Data" e insira uma data qualquer e observe que desta vez foram apresentadas informações referentes ao campo "Data" no log do console, mesmo sem sair do campo;
6. Recarregue a aplicação para realizar um novo teste;
7. Clique ícone "Calendário" do campo "Data" e insira uma data qualquer;
8. Clique no campo "Hora" e **observe que não foi apresentada nenhuma informação no "log" do console**;
9. Clique ícone "Calendário" do campo "Data" e insira uma data qualquer e observe que desta vez foram apresentadas informações referentes ao campo "Data" no log do console, mesmo sem sair do campo;
10. Recarregue a aplicação para realizar um novo teste;
11. Clique no campo "Data" e insira uma data qualquer, **porem via digitação, ou seja, não use o ícone de "Calendário"**;
12. Clique no campo "Hora" e observe que foram apresentadas informações referentes ao campo "Data" no log do console;

[https://stackblitz.com/edit/po-ui-vgonel?devtoolsheight=33&file=src/app/app.component.ts](https://stackblitz.com/edit/po-ui-vgonel?devtoolsheight=33&file=src/app/app.component.ts)
